tests/evaluation/evaluation_formula_array.test.ts

```diff
@@ -740,4 +743,38 @@ describe("evaluate formulas that return an array", () => {
       });
     });
   });
+
+  describe("getSpreadPositionsOf", () => {
+    test("getSpreadPositionsOf getter returns the cells the formula spread on, as well as the cell the formula is on", () => {
+      setCellContent(model, "A1", "=MFILL(2,2,42)");
+      const sheetId = model.getters.getActiveSheetId();
+      expect(model.getters.getSpreadPositionsOf({ sheetId, col: 0, row: 0 })).toEqual([
+        { sheetId, col: 0, row: 0 },
+        { sheetId, col: 0, row: 1 },
+        { sheetId, col: 1, row: 0 },
+        { sheetId, col: 1, row: 1 },
+      ]);
+    });
+
+    test("getSpreadPositionsOf does only return self if the formula could not spread", () => {
+      setCellContent(model, "A1", "=MFILL(2,2,42)");
+      setCellContent(model, "A2", "(ツ)_/¯");
+      expect(model.getters.getSpreadPositionsOf({ sheetId, col: 0, row: 0 })).toEqual([
+        { sheetId, col: 0, row: 0 },
+      ]);
+    });
+
+    test("getSpreadPositionsOf is correct after the evaluation changed so the formula can spread again", () => {
+      setCellContent(model, "H1", "5");
+      setCellContent(model, "A1", "=MFILL(H1,H1,42)");
+
+      expect(model.getters.getSpreadPositionsOf({ sheetId, col: 0, row: 0 })).toHaveLength(25);
+
+      setCellContent(model, "A4", "Block spread");
```

Comment:
I was running this test, trying to debug the evaluation cycle to asses what impacts your changes have. It turns out `"A1"` is now evaluated twice when setting content in "A4", even though it should not. One iteration with [A4, A1] (in that order), then a second iteration with A1 again (which is useless)
I'll check more in details what we should do instead tomorrow.

Reply:
If I understand well, the motivation behind the changes in the evaluation was for `getSpreadPositionsOf` (which was not including the top-left cell).
Could we just change the implementation of `getSpreadPositionsOf` to return what we want without changing anything else to the evaluation? Also, it could actually return a `Zone`.